tools/myst/host/strace.c

```diff
@@ -91,13 +91,73 @@ int myst_set_strace_filter(
     return 0;
 }
 
+int myst_set_strace_id_filter(
+    const char* id_filter,
+    myst_strace_config_t* strace_config,
+    bool tid)
+{
+    char** tokens = NULL;
+    size_t num_tokens = 0;
+    int ret = -1;
+
+    if (myst_strsplit(id_filter, ":", &tokens, &num_tokens) != 0)
+    {
+        fprintf(
+            stderr,
+            "Invalid strace-filter-tid or strace-filter-pid option '%s' "
+            "specified.\n",
+            id_filter);
+        abort();
+    }
+
+    for (size_t i = 0; i < num_tokens; ++i)
+    {
+        long id;
+        const char* _id = tokens[i];
+
+        ret = myst_str2long(_id, &id);
```

Comment:
Should we `ECHECK` here?

Reply:
ECHECK was ignoring the error and causing a bug, so I replaced it with this method